applications/MeshingApplication/tests/test_mpi_parmmg.py

```diff
@@ -0,0 +1,133 @@
+        KratosMultiphysics.DISTANCE,
+        KratosMultiphysics.DISTANCE_GRADIENT,
+        KratosMultiphysics.NODAL_AREA)
+
+        local_gradient.Execute()
+        find_nodal_h = KratosMultiphysics.FindNodalHNonHistoricalProcess(main_model_part)
+        find_nodal_h.Execute()
+
+        ##COMPUTE LEVEL SET METRIC
+        metric_parameters = KratosMultiphysics.Parameters("""
+        {
+            "minimal_size"                             : 1.5,
+            "sizing_parameters": {
+                "reference_variable_name"               : "DISTANCE",
+                "boundary_layer_max_distance"           : 2.0,
+                "interpolation"                         : "constant"
+            },
+            "enforce_current"                      : false,
+            "anisotropy_remeshing"                 : false
+        }
+        """)
+        metric_process = KratosMultiphysics.MeshingApplication.ComputeLevelSetSolMetricProcess3D(
+            main_model_part,
+            KratosMultiphysics.DISTANCE_GRADIENT,
+            metric_parameters)
+        metric_process.Execute()
+
+        ##PERFORM REMESHING
+        pmmg_parameters = KratosMultiphysics.Parameters("""
+        {
+            "filename"                         : "output",
+            "save_external_files"              : true,
+            "save_colors_files"                : true,
+            "initialize_entities"              : false,
+            "preserve_flags"                   : false,
+            "echo_level"                       : 0
+        }
+        """)
+        pmmg_parameters["filename"].SetString(GetFilePath(pmmg_parameters["filename"].GetString()))
+        pmmg_process = KratosMultiphysics.MeshingApplication.ParMmgProcess3D(main_model_part.GetRootModelPart(), pmmg_parameters)
+        pmmg_process.Execute()
+
+        reference_file_name = GetFilePath("parmmg_eulerian_test/cond_ref_map.json")
+        result_file_name = GetFilePath("output_step=0_"+str(communicator.Rank())+".cond.ref.json")
+        self._CompareColorFiles(reference_file_name, result_file_name)
+
+        reference_file_name = GetFilePath("parmmg_eulerian_test/elem_ref_map.json")
+        result_file_name = GetFilePath("output_step=0_"+str(communicator.Rank())+".elem.ref.json")
+        self._CompareColorFiles(reference_file_name, result_file_name)
+
+        ref_mdpa_filename = GetFilePath("parmmg_eulerian_test/parmmg_sphere_reference_mdpa_rank_"+str(communicator.Rank()))
+        ref_model_part = current_model.CreateModelPart("Reference")
+        # We add the variables needed
+        ref_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.DISTANCE)
+        ref_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.DISTANCE_GRADIENT)
+        ref_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.PARTITION_INDEX)
+        ReadSerialModelPart(ref_mdpa_filename, ref_model_part)
```

Comment:
I meant here, why is it necessary to store an entire mdpa for comparing? You anyway only compare the submodelpart names and the number of nodes/elements

I now remembered that I solved this nicely: https://github.com/philbucher/KratosSalomePlugin/blob/master/tests/testing_utilities.py#L515-L572

=> since this is urgent you could copy paste the function here, and after the release move it to the core (didn't think this could be useful for others that why I haven't done it yet)

=> this way you don't have to store any mdpa files and the test will be independent of the number of ranks used!

Reply:
Ok! I still need to have a hierarchy dict for every rank on every world_size possibility no? I agree this is better than storing the whole mdpa tho

Reply:
why do you need it for different sizes?

the number of nodes in totals should be the same no?

Reply:
I think what @philbucher suggest is that you could rewrite the check into something like: 

```
    def _CheckModelPart(self, ref_model_part, result_model_part):
        ref_number_of_node = # (global number of nodes across all partitions), from a json or even hardcoded.
        self.assertEqual(ref_number_of_nodes, communicator.SumAll(NumberOfNodes()))
        # Same with Elems and nodes
```

You end up checking the same but don't need to load the reference mdpa.

Reply:
exactly :)

Reply:
Ok done! Will do merge squash to avoid having the .mdpas there.